pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -3432,11 +3395,89 @@ public void updateArray(String columnName, Array x) throws SQLException {
     } else if (PGobject.class.isAssignableFrom(type)) {
       Object object;
       if (isBinary(columnIndex)) {
-        object = connection.getObject(getPGType(columnIndex), null, this_row[columnIndex - 1]);
+        object = connection.getObject(null, getPGType(columnIndex), null, this_row[columnIndex - 1]);
       } else {
-        object = connection.getObject(getPGType(columnIndex), getString(columnIndex), null);
+        object = connection.getObject(null, getPGType(columnIndex), getString(columnIndex), null);
       }
       return type.cast(object);
+    } else {
+      Map<String, Class<?>> typemap = connection.getTypeMapNoCopy();
+      if (!typemap.isEmpty()) {
+        String pgType = getPGType(columnIndex);
+        Class<?> customType = typemap.get(pgType);
+        if (customType != null) {
```

Comment:
should there be a comparison between the provided `type` and `customType`? the current behavior will result in a ClassCastException, where it seems a SQLException with CANNOT_COERCE state would be better.

Reply:
Fixed.  Thanks.
https://github.com/pgjdbc/pgjdbc/pull/1378/commits/5ff58d39e746a0802ca03a9a9a6ac46a09de9997